### ISSUE TITLE
Transmission 4.05: torrents with colon in name can not be renamed

### ISSUE THREAD

**Issue by zwimer** [transmission/transmission#6419]
### What is the issue?

Despite the merge of https://github.com/transmission/transmission/pull/3823
The following remains an issue: https://github.com/transmission/transmission/issues/1700

Using the WebUI provided on port 9091, attempts to rename torrents with `:`s in their name seemingly do nothing. No error message is displayed or anything, but the torrent does not rename.

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.5

**Comment by zwimer** [comment]
Additionally, having transmission move the torrent's download directory leads to transmission not properly cleaning up the old torrent dir.

For example, telling transmission to move `My: Tor` from `/downloads/A` to /downloads/B` leads to the following:
```
$ find ./downloads/A
./downloads/A/My: Tor/

$ find ./downloads/B
./downloads/B/My: Tor/
./downloads/B/My: Tor/MyFile
```
Here `./downloads/A/My: Tor/` should have been deleted during the move

**Comment by zwimer** [comment]
Additionally, when a download completes and is moved from the `downloading` to the `downloaded` directory, the directory being moved is not properly cleaned up from the downloading directory. For example: If I download the directory `A:B` containing the file `abc` the following will happen:
1. `A:B` will download in `/downloading/A_B`
2. `A:B` will complete downloading.
3. Transmission will move `/downloading/A_B/abc` to `/downloaded/A_B/abc`

Final structure will have:
```
/downloading
/downloading/A_B
/downloaded
/downloaded/A_B
/downloaded/A_B/abc
```

**Comment by tearfur** [comment]
The PR you mentioned wasn't backported to the `4.0.x` branch. The fix will be in `4.1.0`.
